processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -555,19 +592,32 @@ private File makeIndexFiles(
       dimConversions.add(Maps.<String, IntBuffer>newHashMap());
     }
 
+    Map<String, Integer> nullStringIndexMap = Maps.<String, Integer>newHashMap();
```

Comment:
Or we could just *always* include empty string in the `dimensionValues`, and then we don't need this at all, since it'd be OK for MMappedIndexRowIterable to assume that null is 0.